model/common/src/icon4py/model/common/decomposition/mpi_decomposition.py

```diff
@@ -437,3 +439,18 @@ def create_multinode_node_exchange(
         return GHexMultiNodeExchange(props, decomp_info)
     else:
         return SingleNodeExchange()
+
+
+class GlobalReductions(Reductions):
+    def __init__(self, array_ns: ModuleType = np):
+        self._xp = array_ns
+
+    def global_min(self, buffer: data_alloc.NDArray) -> state_utils.ScalarType:
+        if mpi4py is None:
```

Comment:
remove this if and add an implementation of the protocol without the `mpi4py` in `definitions.py`. 
Add a `singledispatch` function to the the return the version base on the `ProcessProperties` implementation. Pretty much the same pattern as for the `definitions.create_exchange` function.

If you run single node (and have no mpi4py installed at all) this file here should not get imported anywhere.